lib/matplotlib/offsetbox.py

```diff
@@ -1389,11 +1389,17 @@ def get_window_extent(self, renderer=None):
         # docstring inherited
         if renderer is None:
             renderer = self.figure._get_renderer()
+        if self.stale:
+            self.update_positions(renderer)
         return Bbox.union([child.get_window_extent(renderer)
                            for child in self.get_children()])
 
     def get_tightbbox(self, renderer=None):
         # docstring inherited
+        if renderer is None:
+            renderer = self.figure._get_renderer()
```

Comment:
With `main`, this method barfs if you don't pass a renderer

```
     ab.get_tightbbox()
  File "/home/ruth/git_stuff/matplotlib/lib/matplotlib/offsetbox.py", line 1397, in get_tightbbox
    return Bbox.union([child.get_tightbbox(renderer)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ruth/git_stuff/matplotlib/lib/matplotlib/offsetbox.py", line 1397, in <listcomp>
    return Bbox.union([child.get_tightbbox(renderer)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ruth/git_stuff/matplotlib/lib/matplotlib/artist.py", line 366, in get_tightbbox
    bbox = self.get_window_extent(renderer)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ruth/git_stuff/matplotlib/lib/matplotlib/offsetbox.py", line 398, in get_window_extent
    renderer = self.figure._get_renderer()
               ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute '_get_renderer'

```

Reply:
Do offset boxes not know their parent Figure?!

Reply:
It looks that way.  Do we need `AnnotationBbox` to have an axes setter similar to `OffsetBox`, where it tells all its children?

https://github.com/matplotlib/matplotlib/blob/e4771f0eb36d01376a39d49b4f5bdb204b06edc7/lib/matplotlib/offsetbox.py#L239-L245

Reply:
https://github.com/matplotlib/matplotlib/blob/2f58e49f8e5feca8eff01e74536842754fe3e4a8/lib/matplotlib/offsetbox.py#L1366-L1371

It _looks_ like `set_figure` on an `AnnotationBbox` _should_ update the figure for both itself _and_ its `offsetbox` attribute. Is that not getting called or set for the AnnotationBbox through means other than calling the method (leaving the offsetbox attribute with it unset)?

Reply:
But if I remove these lines from my branch:

```python
import matplotlib.pyplot as plt
from matplotlib.offsetbox import AnnotationBbox, TextArea

fig, ax = plt.subplots(layout="constrained")

ab = AnnotationBbox(
    TextArea("Some text", textprops={"size": 42}),
    (1, 0.5),
    xycoords="axes fraction",
    box_alignment=(0.5, 0.5),
    pad=0
)

ax.add_artist(ab)
ab.get_tightbbox()
fig.savefig("annotation_box.png", dpi=100)
```

```
    ab.get_tightbbox()
  File "/home/ruth/git_stuff/matplotlib/lib/matplotlib/offsetbox.py", line 1402, in get_tightbbox
    self.update_positions(renderer)
  File "/home/ruth/git_stuff/matplotlib/lib/matplotlib/offsetbox.py", line 1410, in update_positions
    bbox = self.offsetbox.get_bbox(renderer)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ruth/git_stuff/matplotlib/lib/matplotlib/offsetbox.py", line 795, in get_bbox
    _, h_, d_ = renderer.get_text_width_height_descent(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get_text_width_height_descent'

```